apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/ErrorReporting/ErrorReportingTest.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Diagnostics.Common.IntegrationTests;
+using Google.Cloud.Diagnostics.Common.Tests;
+using Google.Cloud.ErrorReporting.V1Beta1;
+using Microsoft.Owin.Testing;
+using Owin;
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Http;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.AspNet.IntegrationTests
+{
+    public class ErrorReportingTest
+    {
+        private readonly ErrorEventEntryPolling _polling = new ErrorEventEntryPolling();
+
+        [Fact]
+        public async Task NoExceptions()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            using (TestServer server = TestServer.Create<ErrorReportingTestApplication>())
+            using (var client = server.HttpClient)
+            {
+                await client.GetAsync($"api/ErrorReporting/Index/{testId}");
+
+                var errorEvents = _polling.GetEvents(startTime, testId, 0);
+                Assert.Empty(errorEvents);
+            }
+        }
+
+        [Fact]
+        public async Task LogsException()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            using (TestServer server = TestServer.Create<ErrorReportingTestApplication>())
+            using (var client = server.HttpClient)
+            {
+                var response = await client.GetAsync($"api/ErrorReporting/ThrowsException/{testId}");
+                var contentTask = response.Content.ReadAsStringAsync();
+
+                Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+
+                var errorEvents = _polling.GetEvents(startTime, testId, 1);
+                Assert.Single(errorEvents);
+                VerifyErrorEvent(errorEvents.First(), testId, "ThrowsException");
```

Comment:
Suggestion:
VerifyErrorEvent(errorEvents.Single(), testId, "ThrowsException");

And you can remove the single check above

Reply:
Done, only I did it in two lines instad of 1

var errorEvent = _polling.GetEvents(startTime, testId, 1).Single();
                VerifyErrorEvent(errorEvent, testId, "ThrowsException");